### PR TITLE
feat: Improve OrderedMap.SetNestedPath

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**

### CODE REVIEW

**Review by michaljurecko — DISMISSED** [review]
LGTM

File: pkg/orderedmap/orderedmap.go [hunk 1]
```diff
@@ -136,20 +152,33 @@ func (o *OrderedMap) SetNestedPath(path Path, value any) error {
...
+					return fmt.Errorf(`slice key can't be negative`)
+				} else if len(s) > int(key) {
 					current = s[key]
 					continue
 				} else {
-					return fmt.Errorf(`path "%s" not found`, currentKey)
+					// Add nil values if the new key isn't immediately after the last
+					for key.Index() > len(s) {
+						s = append(s, nil)
+					}
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply pkg/orderedmap/orderedmap.go:175]: I am not fan of this article but I found it quite some time ago.
https://gist.github.com/GimmyHchs/33bd06e68d72a913a8587b09d41b50d0
I would use `Expand` or `Insert`. Expand seems to be less poluting and inserts new empty slice into slice at given place. I don't know whether it is more effective than for loop but this slice `tricks` looks efficient. Just add comment with given section on `GimmyHchs` github gist and insert slice operation what you want. :slightly_smiling_face:

File: pkg/orderedmap/orderedmap_test.go [hunk 2]
```diff
@@ -363,9 +411,35 @@ func TestOrderedMapSetNested(t *testing.T) {
...
+      ]
     }
   ],
-  "str": "value"
+  "slice2": [
+    4,
+    5,
+    null,
+    null,
+    null,
```
Comment (Matovidlo) [inline pkg/orderedmap/orderedmap_test.go:428]: I would like to know how does it look with `Extend` or `Insert` behaviour that I mentioned above

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19
+          go-version: 1.22
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply .github/workflows/push.yml:11]: No, you can update it.

**Review by michaljurecko — APPROVED** [review]

**Review by Matovidlo — APPROVED** [review]
LGTM

**Pull request merged**